src/runtime/rpc/network.cpp

```diff
@@ -442,6 +457,18 @@ void rpc_session::start_negotiation()
         }
 
         auth_negotiation();
+    } else {
+        // set negotiation success if auth is disabled
+        complete_negotiation(true);
+    }
+}
+
+void rpc_session::complete_negotiation(bool succ)
```

Comment:
I think `on_failure` or `on_success` is easy enough to use, why add a new function complete_negotiation?